### DISCUSSION TITLE
All next/image components fail to load the src due to a 404

### DISCUSSION THREAD

**Discussion by VercammenJens** [vercel/next.js#38386] [Help] [answered]
Hi! 

Today I've updated my nextjs from `12.1.0` to `12.2.0`. Ever since I've been getting errors on my next/image components with a `404 Not Found`. Not using any experimental features like the `next/future/image` but using a CDN (cloudinary). Has there been a breaking change to the next/image component that went over my head? Or does someone know the solution to this problem, or can guide me towards one?

Thanks!

_extra info: still using react 17.0.2, but don't think that should be a problem._

**Comment by icyJoseph** [comment, marked as answer]
Hi,

Has the URL in the images changed? Between 12.1.0 and 12.2.0, that is. I can only think that perhaps the `cloudinary` loader has started to resolve the URLs' differently.

I mean the URL that Next.js resolves to, not the underlying asset URL.

**Reply by icyJoseph** [reply]
Also, you could do a bit of binary search, install 12.1.4, and see if that error still happens, if so, then install 12.1.2, and test again, if 12.1.4, try with 12.1.6, and so on...

**Reply by VercammenJens** [reply]
The URL hasn't changed. If I copy and paste it in another tab it shows the picture. But good thinking, I'll try that version bumping thing to test where the issue is. Thanks for the tip

**Reply by VercammenJens** [reply]
Tried the version bumping to see where the issue is. On 12.1.6 it works, which is the last non-canary build before 12.2.0. Then I tried to check the 12.1.7-canary versions and located the version where the issue began: 12.1.7-canary.32

**Reply by VercammenJens** [reply]
Tried to add the loader to the next.config.js, but nothing really changed:

```
images: {
    domains: [
      'res.cloudinary.com',
      'localhost',
    ],
    loader: 'cloudinary',
    path: 'res.cloudinary.com/api-name'
  },
```

**Reply by icyJoseph** [reply]
mmmm could you add a trailing slash: `'res.cloudinary.com/api-name/'` - So far, I don't see anything that could've triggered this

**Reply by VercammenJens** [reply]
Did that, but I just found the issue. It was inside my `middleware.ts`. I think the pathname of the images has changed which caused it to be redirected to something else. Thank you for helping @icyJoseph

**Reply by icyJoseph** [reply]
Aha yes! Should've mentioned and I should've asked! `/_next` paths now go through middleware.

```ts
import { NextResponse, type NextRequest } from "next/server";

const PUBLIC_FILE = /\.(.*)$/;

export function middleware(req: NextRequest) {
  const { pathname } = req.nextUrl;
  console.log("Middleware", { pathname });

  if (
    pathname.startsWith("/_next") || // exclude Next.js internals
    pathname.startsWith("/api") || //  exclude all API routes
    pathname.startsWith("/static") || // exclude static files
    pathname.includes(".") || // exclude all files in the public folder
    PUBLIC_FILE.test(pathname)
  )
    return NextResponse.next();

  return NextResponse.next();
}
```

I guess that your image paths are going through the middleware as well.

**Reply by VercammenJens** [reply]
Yeah, we had something similar to your solution.
